blaze/expr/collections.py

```diff
@@ -424,10 +424,35 @@ def join(lhs, rhs, on_left=None, on_right=None, how='inner'):
 
 join.__doc__ = Join.__doc__
 
+class IsIn(Expr):
+    """
+    """
+    __slots__ = '_hash', '_child', '_key'
+
+    @property
+    def dshape(self):
+        """ self._child.dshape * number of matched results
```

Comment:
This should be something like

```python
return DataShape(*(self._child.shape + (datashape.bool_,)))
```

The `datashape` programming interface leaves a little to be desired.